app/src/main/java/com/github/polypoly/app/menu/MenuComposable.kt

```diff
@@ -0,0 +1,188 @@
+        ) {
+            var display by remember { mutableStateOf(false) }
+            if(display) {
+                AllButtons()
+            }
+            OptionButton(
+                onClick = { display = !display },
+                iconId = if(display) R.drawable.tmp_sadsmile else R.drawable.tmp_happysmile,
+                description = "display_options"
+            )
+        }
+
+    }
+
+    @Composable
+    private fun AllButtons() {
+        RulesButton()
+        ProfileButton()
+        RankingsButton()
+        SettingsButton()
+    }
+
+    // ===================================================== MENU BUTTONS
+    /**
+     * The button toggles on a dialog component where the rules are displayed.
+     * The rules are scrollable and all the text is stored in RulesObject.
+     *
+     * The dialog box is closed when clicking outside of it
+     */
+    @Composable
+    private fun RulesButton() {
+        var openRules by remember { mutableStateOf(false) }
+        OptionButton(
+            onClick = { openRules = true },
+            iconId = R.drawable.tmp_happysmile,
+            description = "Show Rules"
+        )
+
+        if(openRules) {
+            Dialog(
+                onDismissRequest = { openRules = false },
+            ) {
+                Surface(
+                    color = MaterialTheme.colors.primary,
+                    modifier = Modifier
+                        .fillMaxWidth(0.95f)
+                        .fillMaxHeight(0.95f)
+                ) {
+                    LazyColumn(modifier = Modifier.padding(20.dp)) {
+                        item {
+                            Text(
+                                text = RulesObject.rulesTitle,
+                                style = MaterialTheme.typography.h4
+                            )
+                            Spacer(modifier = Modifier.height(20.dp))
+                        }
+                        items(items = RulesObject.rulesChapters, itemContent = { item ->
+                            Text(
+                                text = item.title,
+                                style = MaterialTheme.typography.h5
+                            )
+                            Text(
+                                text = item.content,
+                                style = MaterialTheme.typography.body1
+                            )
+                            Spacer(modifier = Modifier.height(20.dp))
+                        })
+                    }
+                }
+            }
+        }
+    }
+
+    @Composable
+    private fun ProfileButton() {
+        ActivityOptionButton(
+            destinationActivity = ProfileActivity::class.java,
+            iconId = R.drawable.tmp_happysmile,
+            description = "Open Profile"
+        )
+    }
+
+    @Composable
+    private fun RankingsButton() {
+        ActivityOptionButton(
+            destinationActivity = ProfileActivity::class.java,
```

Comment:
This is supposed to be RankingActivity, not ProfileActivity (or nothing until RankingActivity is done)